aiida/orm/nodes/data/dict.py

```diff
@@ -46,7 +47,7 @@ class Dict(Data):
     Finally, all dictionary mutations will be forbidden once the node is stored.
     """
 
-    def __init__(self, value=None, **kwargs):
+    def __init__(self, value: Union[None, dict, 'Dict'] = None, **kwargs) -> None:
```

Comment:
If you are adding typing, let's use `from __future__ import annotations`
```suggestion
    def __init__(self, value: None | dict | Dict = None, **kwargs) -> None:
```
and why `'Dict'` instead of `Dict`. Can `Dict` not be imported here because of cyclic-imports? If you need to keep it in quotes, best to add
```
from typing import TYPE_CHECKING

if TYPE_CHECKING:
    from aiida.orm import Dict
```

Reply:
The `'Dict'` was mainly due to the fact that the Pylance server I'm running was highlighting it as "'Dict' is not defined", so I found the solution was to use the quotes. But this is no longer necessary since Python 3.7 it seems. And indeed, the actual `pre-commit` run passes fine.

Once I add `from __future__ import annotations` to the top of the file, the Pylance warning is also no longer there.

Reply:
Also wasn't aware of the pipes `|` notation, much cleaner indeed 👌!